pyani_plus/private_cli.py

```diff
@@ -581,7 +586,12 @@ def fastani(  # noqa: PLR0913
                 identity,
                 orthologous_matches,
                 fragments,
-            ) in method_fastani.parse_fastani_file(tmp_output, filename_to_hash)
+            ) in method_fastani.parse_fastani_file(
+                tmp_output,
+                filename_to_hash,
+                # This is used to infer failed alignments:
+                expected_pairs={(_, subject_hash) for _ in query_hashes},
```

Comment:
Here with fastANI, as we are computing one column, the number of expected pairs equals the number of genomes, *n*. That should scale OK.